weasyl/search.py

```diff
@@ -269,18 +280,33 @@ def make_statement(statement_select, statement_additional_where, statement_order
         pagination_filter,
         "ORDER BY content.{{select}} {order} LIMIT %(limit)s".format(order="" if backid else "DESC"))
 
-    params = dict(
-        search_dict,
-        type=type_letter,
-        userid=userid,
-        rating=rating,
-        ratings=list(search.ratings),
-        category=cat,
-        subcategory=subcat,
-        limit=limit,
-        backid=backid,
-        nextid=nextid,
-        required_include_count=len(search.required_includes))
+    all_names = (
+        search.possible_includes |
+        search.required_includes |
+        search.required_excludes)
+
+    tag_ids = searchtag.get_ids(all_names)
+
+    def get_ids(names):
+        return [tag_ids.get(name, -1) for name in names]
+
+    params = {
+        "possible_includes": get_ids(search.possible_includes),
+        "required_includes": get_ids(search.required_includes),
+        "required_excludes": get_ids(search.required_excludes),
+        "required_user_includes": list(search.required_user_includes),
+        "required_user_excludes": list(search.required_user_excludes),
+        "type": type_letter,
+        "userid": userid,
+        "rating": rating,
+        "ratings": list(search.ratings),
+        "category": cat,
+        "subcategory": subcat,
+        "limit": limit,
+        "backid": backid,
+        "nextid": nextid,
+        "required_include_count": len(search.required_includes),
+    }
```

Comment:
Gosh this whole thing makes me sad. Would it be easier to move to query-builder style now? Not that it has to happen in this PR.